openmetadata-service/src/main/java/org/openmetadata/service/util/SubscriptionUtil.java

```diff
@@ -416,42 +418,119 @@ public static Invocation.Builder appendHeadersToTarget(Client client, String uri
+            ? target.put(javax.ws.rs.client.Entity.entity(message, MediaType.APPLICATION_JSON_TYPE))
+            : target.post(
+                javax.ws.rs.client.Entity.entity(message, MediaType.APPLICATION_JSON_TYPE));
 
     LOG.debug(
         "Subscription Destination HTTP Operation {}:{} received response {}",
         httpMethod,
         destination.getSubscriptionDestination().getId(),
         response.getStatusInfo());
 
-    if (response.getStatus() >= 300 && response.getStatus() < 400) {
+    int statusCode = response.getStatus();
+    String statusInfo = response.getStatusInfo().getReasonPhrase();
+    Map<String, List<String>> headers = response.getStringHeaders();
+    String entity = response.hasEntity() ? response.readEntity(String.class) : StringUtils.EMPTY;
+    String mediaType =
+        response.getMediaType() != null ? response.getMediaType().toString() : StringUtils.EMPTY;
+    String location =
+        response.getLocation() != null ? response.getLocation().toString() : StringUtils.EMPTY;
+    long timestamp = System.currentTimeMillis();
+
+    if (statusCode >= 300 && statusCode < 400) {
       // 3xx response/redirection is not allowed for callback. Set the webhook state as in error
-      destination.setErrorStatus(
-          attemptTime, response.getStatus(), response.getStatusInfo().getReasonPhrase());
-    } else if (response.getStatus() >= 400 && response.getStatus() < 600) {
+      handleStatus(
+          destination,
+          TestDestinationStatus.Status.FAILED,
+          flagTestDestination,
+          attemptTime,
+          statusCode,
+          statusInfo,
+          headers,
+          entity,
+          mediaType,
+          location,
+          timestamp,
+          true);
+
+    } else if (statusCode >= 400 && statusCode < 600) {
       // 4xx, 5xx response retry delivering events after timeout
-      destination.setAwaitingRetry(
-          attemptTime, response.getStatus(), response.getStatusInfo().getReasonPhrase());
-    } else if (response.getStatus() == 200) {
-      destination.setSuccessStatus(System.currentTimeMillis());
+      handleStatus(
+          destination,
+          TestDestinationStatus.Status.FAILED,
+          flagTestDestination,
+          attemptTime,
+          statusCode,
+          statusInfo,
+          headers,
+          entity,
+          mediaType,
+          location,
+          timestamp,
+          false);
+
+    } else if (statusCode == 200) {
+      handleStatus(
+          destination,
+          TestDestinationStatus.Status.SUCCESS,
+          flagTestDestination,
+          attemptTime,
+          statusCode,
+          statusInfo,
+          headers,
+          entity,
+          mediaType,
+          location,
+          timestamp,
+          false);
+    }
+  }
+
+  private static void handleStatus(
+      Destination<ChangeEvent> destination,
+      TestDestinationStatus.Status testStatus,
+      boolean flagTestDestination,
```

Comment:
@Siddhanttimeline can't we just handle the status for test in different method, we shouldn't need to pass falg around